pipert/core/pipeline_manager.py

```diff
@@ -0,0 +1,441 @@
+        del self.components[component_name]
+        return self._create_response(
+            True,
+            f"Component {component_name} has been removed"
+        )
+
+    @component_name_existence_error(need_to_be_exist=True)
+    def add_routine_to_component(self, component_name,
+                                 routine_type_name, **routine_parameters_kwargs):
+        if self._does_component_running(self.components[component_name]):
+            return self._create_response(
+                False,
+                "You can't add a routine while your component is running"
+            )
+
+        routine_class_object = self._get_routine_class_object_by_type_name(routine_type_name)
+
+        if routine_class_object is None:
+            return self._create_response(
+                False,
+                f"The routine type '{routine_type_name}' doesn't exist"
+            )
+
+        if "name" not in routine_parameters_kwargs:
+            return self._create_response(
+                False,
+                f"Routine must have a name"
+            )
+
+        if self.components[component_name] \
+                .does_routine_name_exist(routine_parameters_kwargs["name"]):
+            return self._create_response(
+                False,
+                f"Routine with the name {routine_parameters_kwargs['name']}"
+                f" already exist in this component"
+            )
+
+        try:
+            # replace all queue names with the queue objects of the component before creating routine
+            for key, value in routine_parameters_kwargs.items():
+                if 'queue' in key.lower():
+                    routine_parameters_kwargs[key] = self.components[component_name] \
+                        .get_queue(queue_name=value)
+
+            routine_parameters_kwargs["component_name"] = component_name
+
+            self.components[component_name] \
+                .register_routine(routine_class_object(**routine_parameters_kwargs)
+                                  .as_thread())
+            return self._create_response(
+                True,
+                f"The routine {routine_parameters_kwargs['name']} has been added"
+            )
+        except QueueDoesNotExist as e:
+            return self._create_response(
+                False,
+                e.message()
+            )
+        except TypeError as error:
+            return self._create_response(
+                False,
+                str(error)
+            )
+
+    @component_name_existence_error(need_to_be_exist=True)
+    def remove_routine_from_component(self, component_name, routine_name):
+        if self._does_component_running(self.components[component_name]):
+            return self._create_response(
+                False,
+                "You can't remove a routine while your component is running"
+            )
+        self.components[component_name].remove_routine(routine_name)
```

Comment:
if routine does not exist?

Reply:
The method removes any routine with the given name, it doesn't return any indication whether the method deleted or not because the routines stored as a list inside the component.
I can rewrtie the function to run over the list and return the True if deleted a routine and False if it didn't.